docker/k8s-glue/Dockerfile

```diff
@@ -0,0 +1,41 @@
+ARG TAG=3.7.12-alpine3.15
+
+FROM python:${TAG} as build
+
+RUN apk add --no-cache \
+    gcc \
+    musl-dev \
+    libffi-dev
+
+RUN python3 \
```

Comment:
Why not use `setup.sh` as we do [here](https://github.com/allegroai/clearml-agent/blob/d39e30995a0d0be14ee266a8375d81381c85548c/docker/k8s-glue/glue-build-aws/Dockerfile#L12)? Since we have several Dockerfile flavors, I really prefer to keep the non-cloud-specific installations in one place 🙂 

Also, I agree the `--no-cache` is important (and can be added there). but why is the `--prefix` switch needed?

Reply:
`setup.sh` and `entrypoint.sh` and any other shell scripts are making the whole thing more complex and harder to understand "what is happening here?". I spent half a day understanding the purpose of shell files and didn't find anything that couldn\t be done by Dockerfile, so idea is to make it as plain and simple as possible. 

Now about the `--prefix`. As you can see I'm using Docker multistage build here (multiple FROM entries). The first stage requires development dependencies to build python wheels, and I want a predictable location with built files, so that I set `--prefix` parameter. In the second stage (the runtime) I just copy the files from the first stage, but the image itself won't contain dev dependencies and will be smaller in size.

Reply:
I totally understand your point about simplicity, but what I'm worried about is that when we fix versions and dependencies for AWS or GCP builds, we might miss updating them in the new Dockerfile - duplication usually lead to bugs 🙁

Reply:
Correct me if I'm wrong. GCP and AWS Dockerfiles contain gcloud and aws-cli (and aws-iam-authenticator) additionally to the k8s-glue. So that we can use Dockerfile from this PR as a base and install google/aws tools in the nested Dockerfiles.

Reply:
That's possible, but would require more changes and testing

Reply:
Would you like me to add these changes to this PR?

Reply:
If you can, that would be very cool :)

Reply:
Alright, next iteration. Docker complained about files outside its context, so I placed `k8s_glue_example.py` to `glue-build` dir. 

Prepared 2 Dockerfiles:
- [Dockerfile.alpine](https://github.com/allegroai/clearml-agent/blob/db5ae8aeaafc47b5f24c68e7e56406699c6a38d2/docker/k8s-glue/glue-build/Dockerfile.alpine) - based on Alpine Linux, intended to be small.  The file includes stages to build k8s-glue and Google Cloud SDK. Unfortunately, AWS CLI v2 can't be installed easily, so I skipped that and placed comments, hope they finish 'build from source' soon.
- [Dockerfile.bullseye](https://github.com/allegroai/clearml-agent/blob/db5ae8aeaafc47b5f24c68e7e56406699c6a38d2/docker/k8s-glue/glue-build/Dockerfile.bullseye) - based on debian bullseye, the file includes stages to build k8s-glue, AWS CLI v2 and Google Cloud SDK. 

Use these commands to build the image with tools you require (run it from glue-build folder or change build context accordingly):

```
sudo DOCKER_BUILDKIT=1 docker build . -f Dockerfile.alpine --target=target # builds k8s-glue only
sudo DOCKER_BUILDKIT=1 docker build . -f Dockerfile.alpine --target=gcp # builds k8s-glue and Google Cloud SDK
sudo DOCKER_BUILDKIT=1 docker build . -f Dockerfile.bullseye --target=target # builds k8s-glue
sudo DOCKER_BUILDKIT=1 docker build . -f Dockerfile.bullseye --target=aws # builds k8s-glue and AWS CLI v2
sudo DOCKER_BUILDKIT=1 docker build . -f Dockerfile.bullseye --target=gcp # builds k8s-glue and Google Cloud SDK
```

Reply:
Hi @xadcoh, sorry for the delay... this looks cool, thanks! 🙏